test/utils/format/formatSmart.spec.ts

```diff
@@ -0,0 +1,184 @@
+  })
+})
+
+describe('Tokens with precision 0', () => {
+  test('1 unit', async () => {
+    const amount = new BN('1')
+    expect(formatSmart(amount, 0)).toEqual('1')
+  })
+
+  test('12,345 units', async () => {
+    const amount = new BN('12345')
+    expect(formatSmart(amount, 0)).toEqual('12,345')
+  })
+
+  test('4,567,890 units', async () => {
+    const amount = new BN('4567890')
+    expect(formatSmart(amount, 0)).toEqual('4,567,890')
+  })
+})
+
+describe('Tokens with precision 2', () => {
+  test('1 unit', async () => {
+    const amount = new BN('100')
+    expect(formatSmart(amount, 2)).toEqual('1')
+  })
+
+  test('12.34 units', async () => {
+    const amount = new BN('1234')
+    expect(formatSmart(amount, 2)).toEqual('12.34')
+  })
+
+  test('1,234,567.89 units', async () => {
+    const amount = new BN('123456789')
+    expect(formatSmart(amount, 2)).toEqual('1,234,567.89')
+  })
+
+  test('1,234,567.8 units', async () => {
+    const amount = new BN('123456780')
+    expect(formatSmart(amount, 2)).toEqual('1,234,567.8')
+  })
+})
+
+describe('2 decimals', () => {
+  test('1 unit', async () => {
+    const amount = new BN('1000000')
+    expect(formatSmart({ amount, precision: 6, decimals: 2 })).toEqual('1')
+  })
+
+  test('12.34 units', async () => {
+    const amount = new BN('12340000')
+    expect(formatSmart({ amount, precision: 6, decimals: 2 })).toEqual('12.34')
+  })
+
+  test('4,567,890.123333 units, round down', async () => {
+    const amount = new BN('4567890123333')
+    expect(formatSmart({ amount, precision: 6, decimals: 2 })).toEqual('4,567,890.12')
+  })
+
+  test('4,567,890.125555 units, round up', async () => {
+    const amount = new BN('4567890125555')
+    expect(formatSmart({ amount, precision: 6, decimals: 2 })).toEqual('4,567,890.13')
+  })
+})
+
+describe('0 decimals', () => {
+  test('1 unit', async () => {
+    const amount = new BN('1000000')
+    expect(formatSmart({ amount, precision: 6, decimals: 0 })).toEqual('1')
+  })
+
+  test('12.34 units', async () => {
+    const amount = new BN('12340000')
+    expect(formatSmart({ amount, precision: 6, decimals: 0 })).toEqual('12')
+  })
+
+  test('4,567,890.123333 units, round down', async () => {
+    const amount = new BN('4567890123333')
+    expect(formatSmart({ amount, precision: 6, decimals: 0 })).toEqual('4,567,890')
+  })
+
+  test('4,567,890.125555 units, round up', async () => {
+    const amount = new BN('4567890125555')
+    expect(formatSmart({ amount, precision: 6, decimals: 0 })).toEqual('4,567,890')
+  })
+})
+
+describe('Big amounts', () => {
+  // TODO: Considering showing K,M,B,...
+  test('1B Ether', async () => {
+    expect(formatSmart(new BN(toWei(new BN('1000000000'), 'ether')), DEFAULT_PRECISION)).toEqual('1B')
+  })
+
+  // TODO: Define what for arbitrarily big amounts
+  test('uint max value', async () => {
+    const expected = '115,792,089,237,316,195,423,570,985,008,687,907,853,269,984,665.64T'
```

Comment:
May I suggest we return this instead?

```suggestion
    const expected = 'A very big number. Trust me, it\'s THAT big.'
```

Reply:
that amount is my ETH goals